Source/DafnyCore/DafnyFile.cs

```diff
@@ -42,127 +47,157 @@ public class DafnyFile {
       }
-      return result;
+      yield break;
     }
 
-    var filePath = uri.LocalPath;
-
-    origin ??= Token.NoToken;
+    uriOrigin ??= Token.NoToken;
 
-    string canonicalPath;
-    string baseName;
-    var extension = DafnyFileExtension;
+    string extension;
     if (uri.IsFile) {
       extension = Path.GetExtension(uri.LocalPath).ToLower();
-      baseName = Path.GetFileName(uri.LocalPath);
-      // Normalizing symbolic links appears to be not
-      // supported in .Net APIs, because it is very difficult in general
-      // So we will just use the absolute path, lowercased for all file systems.
-      // cf. IncludeComparer.CompareTo
-      canonicalPath = Canonicalize(uri.LocalPath).LocalPath;
     } else if (uri.Scheme == "dllresource") {
       extension = Path.GetExtension(uri.LocalPath).ToLower();
-      baseName = uri.LocalPath;
-      canonicalPath = uri.ToString();
     } else {
-      canonicalPath = "";
-      baseName = "";
-    }
-
-    if (uri.Scheme == "stdin") {
-      return HandleStandardInput(options, uri, origin);
+      extension = DafnyFileExtension;
     }
 
     if (uri.Scheme == "untitled" || extension == DafnyFileExtension || extension == ".dfyi") {
-      return HandleDafnyFile(options, fileSystem, reporter,
-        uri, origin, asLibrary, extension, canonicalPath, baseName, filePath);
+      var file = HandleDafnyFile(fileSystem, reporter, options, uri, uriOrigin, asLibrary);
+      if (file != null) {
+        yield return file;
+      }
+      yield break;
     }
 
     if (extension == DooFile.Extension) {
-      return await HandleDooFile(options, fileSystem,
-        reporter, uri, origin, asLibrary);
+      await foreach (var dooResult in HandleDooFile(fileSystem, reporter, options, uri, uriOrigin, asLibrary)) {
+        yield return dooResult;
+      }
+      yield break;
     }
 
     if (extension == ".dll") {
-      return HandleDll(options, uri, origin, filePath, extension, canonicalPath, baseName);
+      var dllResult = HandleDll(options, uri, uriOrigin);
+      if (dllResult != null) {
+        yield return dllResult;
+      }
+      yield break;
     }
 
     var handler = ExtensionHandlers.GetValueOrDefault(extension);
     if (handler != null) {
-      return await handler(options, fileSystem, reporter, uri, origin, asLibrary);
+      await foreach (var result in handler(options, fileSystem, reporter, uri, uriOrigin, asLibrary)) {
+        yield return result;
+      }
+      yield break;
     }
     if (errorOnNotRecognized != null) {
       reporter.Error(MessageSource.Project, Token.Cli, errorOnNotRecognized);
     }
-    return null;
   }
 
-  private static DafnyFile HandleDafnyFile(DafnyOptions options, IFileSystem fileSystem,
+  public static readonly Option<bool> UnsafeDependencies = new("--unsafe-dependencies",
+    "Allows Dafny to accept dependencies that may not have been verified with compatible options, which can be useful during development.");
```

Comment:
```suggestion
    "Allows Dafny to accept dependencies that may not have been previously verified, which can be useful during development.");
```

Or something like that. The other wording implies the options may not be compatible, but AFAICT we're explicitly checking that now, just not actually verifying the dependencies.